pkg/cgroups/cgroups_linux.go

```diff
@@ -45,6 +49,51 @@ func mountCgroup() error {
 	return nil
 }
 
+// cgrpSanityCheck performs a sanity check on the mounted cgroup fs and issues
+// a warning if something seems wrong.
+//
+// It does this by recursively checking the number of pids in "cgroup.procs". If they are 2
+// or less, it is likely that the cilium pod is the only container in this
+// cgroup.
+func cgrpSanityCheck() {
+
+	procCount := 0
+	err := filepath.Walk(cgroupRoot, func(path string, info fs.FileInfo, err error) error {
+		if err != nil {
+			log.WithField("path", path).WithError(err).Info("cgrpSanityCheck: error traversing cgroup directory")
+		}
+
+		if info.IsDir() {
+			return nil
+		}
+
+		if info.Name() != "cgroup.procs" {
+			return nil
+		}
+
+		f, err := os.Open(path)
+		if err != nil {
+			log.WithField("path", path).WithError(err).Info("cgrpSanityCheck: failed to open file")
+			return nil
+		}
+		s := bufio.NewScanner(f)
+		for s.Scan() {
+			procCount++
+			if procCount > 2 {
```

Comment:
Should the value of `2` be in a const, and documented?

Reply:
IMO, this is the most readable way to write this code. For example, having this value defined outside of the function as a const makes the code less readable.

Reply:
OK, makes sense. Your comment for the function explains why we compare with `2`, so it should be enough, let's keep as it is.